admin interface for translation

danielzhang130 opened the pull request:
---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

dependabot[bot] committed:
Bump eslint from 7.29.0 to 7.30.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.29.0 to 7.30.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.29.0...v7.30.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

danielzhang130 committed:
- admin interface for translation
- ability to sort by most used
- ability to delete old entries manually

crowdin-bot committed:
New Crowdin translations by Github Action

HCrane committed:
Merge pull request #1791 from Catrobat/l10n_crowdin_action

New Crowdin translations

dependabot[bot] committed:
Bump lexik/jwt-authentication-bundle from 2.12.1 to 2.12.3

Bumps [lexik/jwt-authentication-bundle](https://github.com/lexik/LexikJWTAuthenticationBundle) from 2.12.1 to 2.12.3.
- [Release notes](https://github.com/lexik/LexikJWTAuthenticationBundle/releases)
- [Changelog](https://github.com/lexik/LexikJWTAuthenticationBundle/blob/2.x/CHANGELOG.md)
- [Commits](https://github.com/lexik/LexikJWTAuthenticationBundle/compare/v2.12.1...v2.12.3)

---
updated-dependencies:
- dependency-name: lexik/jwt-authentication-bundle
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

JoshuaSchaefer commented on src/Admin/Translation/Controller/MachineTranslationAdminController.php:
```diff
@@ -0,0 +1,57 @@
+      return new RedirectResponse($this->admin->generateUrl('list'));
+    }
+
+    $entity = $this->type == self::TYPE_PROJECT ? '--only-project' : '--only-comment';
```
Should this maybe be if ($this->type === self::TYPE_PROJECT) {} else if ($this->type === self::TYPE_COMMENT) {} ? If this is ever expanded to include more sections, then this might be missed in the update and cause a bug or something since it really only handles the two cases with this type of statement. Another option would be to create some kind of map or dictionary similar to what we did with the different languages and providers on the UI.

JoshuaSchaefer commented:
Should there also be any unit or behat tests added for the controllers and screen or is that overkill for the admin actions?

dependabot[bot] committed:
Bump external-svg-loader from 1.3.2 to 1.3.3

Bumps [external-svg-loader](https://github.com/shubhamjain/svg-loader) from 1.3.2 to 1.3.3.
- [Release notes](https://github.com/shubhamjain/svg-loader/releases)
- [Commits](https://github.com/shubhamjain/svg-loader/commits)

---
updated-dependencies:
- dependency-name: external-svg-loader
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump php-http/httplug-bundle from 1.20.1 to 1.21.0

Bumps [php-http/httplug-bundle](https://github.com/php-http/HttplugBundle) from 1.20.1 to 1.21.0.
- [Release notes](https://github.com/php-http/HttplugBundle/releases)
- [Changelog](https://github.com/php-http/HttplugBundle/blob/master/CHANGELOG.md)
- [Commits](https://github.com/php-http/HttplugBundle/compare/1.20.1...1.21.0)

---
updated-dependencies:
- dependency-name: php-http/httplug-bundle
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump shivammathur/setup-php from 2.11.0 to 2.12.0

Bumps [shivammathur/setup-php](https://github.com/shivammathur/setup-php) from 2.11.0 to 2.12.0.
- [Release notes](https://github.com/shivammathur/setup-php/releases)
- [Commits](https://github.com/shivammathur/setup-php/compare/2.11.0...2.12.0)

---
updated-dependencies:
- dependency-name: shivammathur/setup-php
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump npm from 7.19.1 to 7.20.0

Bumps [npm](https://github.com/npm/cli) from 7.19.1 to 7.20.0.
- [Release notes](https://github.com/npm/cli/releases)
- [Changelog](https://github.com/npm/cli/blob/latest/CHANGELOG.md)
- [Commits](https://github.com/npm/cli/compare/v7.19.1...v7.20.0)

---
updated-dependencies:
- dependency-name: npm
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump phpstan/phpstan-symfony from 0.12.38 to 0.12.39

Bumps [phpstan/phpstan-symfony](https://github.com/phpstan/phpstan-symfony) from 0.12.38 to 0.12.39.
- [Release notes](https://github.com/phpstan/phpstan-symfony/releases)
- [Commits](https://github.com/phpstan/phpstan-symfony/compare/0.12.38...0.12.39)

---
updated-dependencies:
- dependency-name: phpstan/phpstan-symfony
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

crowdin-bot committed:
New Crowdin translations by Github Action

HCrane committed:
Merge pull request #1808 from Catrobat/dependabot/github_actions/shivammathur/setup-php-2.12.0

Bump shivammathur/setup-php from 2.11.0 to 2.12.0

HCrane committed:
Merge pull request #1811 from Catrobat/dependabot/composer/phpstan/phpstan-symfony-0.12.39

Bump phpstan/phpstan-symfony from 0.12.38 to 0.12.39

HCrane committed:
Merge pull request #1802 from Catrobat/l10n_crowdin_action

New Crowdin translations

HCrane committed:
Merge pull request #1806 from Catrobat/dependabot/composer/php-http/httplug-bundle-1.21.0

Bump php-http/httplug-bundle from 1.20.1 to 1.21.0

HCrane committed:
Merge pull request #1810 from Catrobat/dependabot/npm_and_yarn/npm-7.20.0

Bump npm from 7.19.1 to 7.20.0

HCrane committed:
Merge pull request #1799 from Catrobat/dependabot/composer/lexik/jwt-authentication-bundle-2.12.3

Bump lexik/jwt-authentication-bundle from 2.12.1 to 2.12.3

HCrane committed:
Merge pull request #1794 from Catrobat/dependabot/npm_and_yarn/eslint-7.30.0

Bump eslint from 7.29.0 to 7.30.0

HCrane committed:
Merge pull request #1805 from Catrobat/dependabot/npm_and_yarn/external-svg-loader-1.3.3

Bump external-svg-loader from 1.3.2 to 1.3.3

danielzhang130 committed:
- Remove translation button for projects and comments of the user owning it
- Behat tests for translation button visibility
- Remove verification for visibility embedded in other test cases (test one thing in each test)

dependabot[bot] committed:
Bump phpstan/phpstan-symfony from 0.12.39 to 0.12.41

Bumps [phpstan/phpstan-symfony](https://github.com/phpstan/phpstan-symfony) from 0.12.39 to 0.12.41.
- [Release notes](https://github.com/phpstan/phpstan-symfony/releases)
- [Commits](https://github.com/phpstan/phpstan-symfony/compare/0.12.39...0.12.41)

---
updated-dependencies:
- dependency-name: phpstan/phpstan-symfony
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump eslint from 7.30.0 to 7.31.0

Bumps [eslint](https://github.com/eslint/eslint) from 7.30.0 to 7.31.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v7.30.0...v7.31.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump stylelint-config-twbs-bootstrap from 2.2.1 to 2.2.2

Bumps [stylelint-config-twbs-bootstrap](https://github.com/twbs/stylelint-config-twbs-bootstrap) from 2.2.1 to 2.2.2.
- [Release notes](https://github.com/twbs/stylelint-config-twbs-bootstrap/releases)
- [Commits](https://github.com/twbs/stylelint-config-twbs-bootstrap/compare/v2.2.1...v2.2.2)

---
updated-dependencies:
- dependency-name: stylelint-config-twbs-bootstrap
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

crowdin-bot committed:
New Crowdin translations by Github Action

HCrane committed:
Merge pull request #1814 from danielzhang130/my_program_translate

Remove translation button for projects and comments that the user owns

HCrane committed:
Merge pull request #1813 from Catrobat/l10n_crowdin_action

New Crowdin translations

HCrane committed:
Merge pull request #1818 from Catrobat/dependabot/npm_and_yarn/stylelint-config-twbs-bootstrap-2.2.2

Bump stylelint-config-twbs-bootstrap from 2.2.1 to 2.2.2

HCrane committed:
Merge pull request #1817 from Catrobat/dependabot/npm_and_yarn/eslint-7.31.0

Bump eslint from 7.30.0 to 7.31.0

HCrane committed:
Merge pull request #1816 from Catrobat/dependabot/composer/phpstan/phpstan-symfony-0.12.41

Bump phpstan/phpstan-symfony from 0.12.39 to 0.12.41

HCrane committed:
Merge pull request #1797 from danielzhang130/translate_db_admin

Add listener and database for translation
